### PR TITLE
Min sdk15

### PR DESCRIPTION
**Pull request by Akhil-Tripathi**
I have made the bare minimum changes to make it compatible with sdk versions as low as 15. @michaelprimez  Thanks for your base repo.

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <entry name="!?*.clj" />
+      <entry name="!?*.aj" />
+    </wildcardResourcePatterns>
+    <annotationProcessing>
+      <profile default="true" name="Default" enabled="false">
+        <processorPath useClasspath="true" />
+      </profile>
+    </annotationProcessing>
+  </component>
+</project>
```
Comment (michaelprimez) [inline .idea/compiler.xml]: Please remove IDE specific file from the pr.

File: .idea/copyright/profiles_settings.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
+  <settings default="" />
```
Comment (michaelprimez) [inline .idea/copyright/profiles_settings.xml]: Please remove IDE specific file from the pr.

File: .idea/gradle.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline .idea/gradle.xml]: Please remove IDE specific file from the pr.

File: .idea/misc.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline .idea/misc.xml]: Please remove IDE specific file from the pr.

File: .idea/modules.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline .idea/modules.xml]: Please remove IDE specific file from the pr.

File: .idea/runConfigurations.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline .idea/runConfigurations.xml]: Please remove IDE specific file from the pr.

File: .idea/vcs.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline .idea/vcs.xml]: Please remove IDE specific file from the pr.

File: build.gradle [hunk 8]
```diff
@@ -17,6 +17,10 @@ allprojects {
     repositories {
```
Comment (michaelprimez) [inline build.gradle]: Please pull master and resolve conflicts before the PR

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -0,0 +1,6 @@
+#Sat Nov 11 16:39:53 IST 2017
```
Comment (michaelprimez) [inline gradle/wrapper/gradle-wrapper.properties]: Please remove gradle-wrapper.properties file from the PR.

File: gradlew [hunk 10]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
```
Comment (michaelprimez) [inline gradlew]: Please remove gradlew file from the PR.

File: gradlew.bat [hunk 11]
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (michaelprimez) [inline gradlew.bat]: Please remove gradlew.bat file from the PR.

File: searchablespinner/.idea/compiler.xml [hunk 12]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/compiler.xml]: Please remove compiler.xml file from the PR.

File: searchablespinner/.idea/copyright/profiles_settings.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
```
Comment (michaelprimez) [inline searchablespinner/.idea/copyright/profiles_settings.xml]: Please remove profiles_settings.xml file from the PR.

File: searchablespinner/.idea/gradle.xml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/gradle.xml]: Please remove gradle.xml file from the PR.

File: searchablespinner/.idea/misc.xml [hunk 15]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/misc.xml]: Please remove misc.xml file from the PR.

File: searchablespinner/.idea/modules.xml [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/modules.xml]: Please remove modules.xml file from the PR.

File: searchablespinner/.idea/runConfigurations.xml [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/runConfigurations.xml]: Please remove runConfigurations.xml file from the PR.

File: searchablespinner/.idea/workspace.xml [hunk 18]
```diff
@@ -0,0 +1,247 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelprimez) [inline searchablespinner/.idea/workspace.xml]: Please remove workspace.xml file from the PR.

File: searchablespinner/gradle/wrapper/gradle-wrapper.properties [hunk 19]
```diff
@@ -0,0 +1,6 @@
+#Sat Nov 11 16:42:35 IST 2017
```
Comment (michaelprimez) [inline searchablespinner/gradle/wrapper/gradle-wrapper.properties]: Please remove gradle-wrapper.properties file from the PR.

File: searchablespinner/gradlew [hunk 20]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
```
Comment (michaelprimez) [inline searchablespinner/gradlew]: Please remove gradlew file from the PR.

File: searchablespinner/gradlew.bat [hunk 21]
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (michaelprimez) [inline searchablespinner/gradlew.bat]: Please remove gradlew file from the PR.

File: searchablespinner/local.properties [hunk 22]
```diff
@@ -0,0 +1,11 @@
+## This file is automatically generated by Android Studio.
```
Comment (michaelprimez) [inline searchablespinner/local.properties]: Please remove local.properties file from the PR.

**Review by michaelprimez — CHANGES_REQUESTED** [review]
@Akhil-Tripathi thanks for the PR. Please remove any IDE related file and pull the latest master before you push your changes.

**Comment by Akhil-Tripathi** [comment]
@michaelprimez removed ide induced changes

File: app/build.gradle [hunk 23]
```diff
@@ -5,7 +5,7 @@ android {
     buildToolsVersion '26.0.2'
     defaultConfig {
         applicationId "gr.escsoft.michaelprimez.searchablespinnerexamples"
-        minSdkVersion 21
+        minSdkVersion 19
```
Comment (michaelprimez) [inline app/build.gradle]: Why 19? Please use minSdkVersion 15 and try run it in an emulator. You'll see some fixes you have to make related to ViewTreeObserver, actually the app is crashing with "java.lang.NoSuchMethodError: android.view.ViewTreeObserver.removeOnGlobalLayoutListener". Also to support older version than API 17 you should add paddingRight, paddingLeft whenever a paddingStart, paddingEnd is used.

**Review by michaelprimez — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 23 above)
Comment (Akhil-Tripathi) [reply app/build.gradle]: Hey I did make the library minimum sdk 15. If u need to make the demo app suitable for <19 then please make the needed changes. I am using the lib in one of my target projects and it's working fine over there. The pr purpose was to make the lib support sdk as low as 15

Comment (michaelprimez) [reply app/build.gradle]: And what you did about the "NoSuchMethodError"? did you remove it? And what you did about paddingStart, paddingEnd? Did you fix it in the instance of the library that you use in your projects? If yes, why don't you fix it in the pr in order to help others? It is not so polite to make an uncompleted pr and then ask for other to make fixes. Probably in your projects you don't use min sdk 15 but the title of the pr is "Min sdk15" after all. I don't understand you at all...

**Comment by Akhil-Tripathi** [comment]
Hey, about the padding start and end‎ case I did place a min sdk check if I remember. I did not face the NoSuchMethod error. The purpose of this pr itself is to show others how to use the lib for lower sdk versions. I can attend to the NoSuchMethod error in a few days, won't be possible to attend to it straight away.                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: Michael KeskinidisSent: Saturday 25 November 2017 18:06To: michaelprimez/searchablespinnerReply To: michaelprimez/searchablespinnerCc: Akhil Tripathi; MentionSubject: Re: [michaelprimez/searchablespinner] Min sdk15 (#20)‎@michaelprimez commented on this pull request.



In app/build.gradle:
> @@ -5,7 +5,7 @@ android {
     buildToolsVersion '26.0.2'
     defaultConfig {
         applicationId "gr.escsoft.michaelprimez.searchablespinnerexamples"
-        minSdkVersion 21
+        minSdkVersion 19

And what you did about the "NoSuchMethodError"? did you remove it? And what you did about paddingStart, paddingEnd? Did you fix it in the instance of the library that you use in your projects? If yes, why don't you fix it in the pr in order to help others? It is not so polite to make an uncompleted pr and then ask for other to make fixes. Probably in your projects you don't use min sdk 15 but the title of the pr is "Min sdk15" after all. I don't understand you at all...

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/michaelprimez/searchablespinner","title":"michaelprimez/searchablespinner","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/michaelprimez/searchablespinner"}},"updates":{"snippets":[{"icon":"PERSON","message":"@michaelprimez commented on #20"}],"action":{"name":"View Pull Request","url":"https://github.com/michaelprimez/searchablespinner/pull/20#discussion_r153045072"}}}

**Pull request closed without merging**